### PR TITLE
Add support for multiple testsuites in junit xml

### PR DESCRIPTION
**Pull request by luispadron**
If I have two `.xcresult` files and I run:

- `xcrun xcresulttool merge Result1.xcresult Result2.xcresult --output-path=Result_merged.xcresult`
- `xcresultparser Result_merged.xcresult --output-format=junit`

The resulting junit file will contain only a single `testsuite` even when the xcresult specifies there are multiple.

This PR updates the parser to:

- Iterate through all available test actions
- Create a test suite for each test action
- Set the overall test time of the `testsuites` to the sum of all test suite times.
- Adds a test for this merged example

### CODE REVIEW

File: Sources/xcresultparser/JunitXML.swift [hunk 1]
```diff
@@ -72,56 +72,69 @@ public struct JunitXML: XmlSerializable {
...
-            for thisTestableSummary in thisSummary.testableSummaries {
-                for group in thisTestableSummary.tests {
-                    for testsuite in createTestSuite(group, failureSummaries: failureSummaries) {
-                        testsuites.addChild(testsuite)
+
+        var overallTestSuiteDuration = 0.0
+        for testAction in testActions {
+            guard let testsId = testAction.actionResult.testsRef?.id,
+                  let testPlanRun = resultFile.getTestPlanRunSummaries(id: testsId) else {
+                      return xml.xmlString(options: [.nodePrettyPrint, .nodeCompactEmptyElement])
```
Comment (a7ex) [inline Sources/xcresultparser/JunitXML.swift:106]: Shouldn't we rather "continue" here.
Maybe it's only this loop iteration and the next loop iteration is fine?

**Review by a7ex — COMMENTED** [review]
Good addition, thank you very much for your contribution!
But I think, we might handle the loop more graceful.

File: Sources/xcresultparser/JunitXML.swift (hunk 1 above)
Comment (luispadron) [reply Sources/xcresultparser/JunitXML.swift:106]: Oh yeah definitely, I'll get it updated

Comment (luispadron) [reply Sources/xcresultparser/JunitXML.swift]: Updated

**Review by a7ex — APPROVED** [review]

**Pull request merged**